### PR TITLE
feat: add sanity swap checks after handshake

### LINKED ISSUE #676: Sanity test swaps
This is an offshoot from discussion is #661. We might want to do a test swap (1 satoshi for 1 satoshi, or maybe even 1 msat for 1 msat if possible) for each trading pair that is active with a peer (or maybe two test swaps, with each side sending and receiving both pairs). This will verify that routes exist and lnd is properly set up on both sides, and rule out a lot of the problems that might cause swaps to fail.

I'm thinking that this can be part of the handshake process when we connect to a peer. @offerm asked whether we should do it each time we get an order from a peer. I think that could possibly be a little over the top, as we could easily wind up increasing the number of swaps we're making by an order of magnitude or two if we're connected to a lot of peers. I think the benefit is also pretty limited, particularly if we've just had a successful test swap a few seconds earlier. #675 though proposes that we do a `queryroutes` check for each order we receive, which I think is less burdensome and also possibly more helpful since we can see if a route exists for large orders which we don't check with a 1 satoshi test swap.

### PR DESCRIPTION
**Pull request by sangaman**
Closes #676.

This is a major commit that changes the procedure for accepting orders from peers to ensure that we can send and receive payments for both currencies that make up the order's trading pair. This introduces the concept of "sanity swaps," which are atomic swaps of 1 satoshi for 1 satoshi within a single currency/chain. The flow for atomic swaps is as follows:

1. A connection is established with a peer.
2. The handshake is performed successfully.
3. A `SanitySwapPacket` will be sent to the peer to inform them of a sanity swap attempt for each currency the appears in the peer's trading pairs advertised in the `nodeState`.
4. We send a 1 satoshi payment to the peer for each sanity swap.
5. Upon receiving our 1 satoshi payment, the peer attempts to learn the preimage by sending a 1 satoshi payment to us.
6. We release the preimage that we created, effectively settling the swap.
7. After all sanity swaps resolve, we check which advertised pairs had successful swaps for both the base and quote currency.
8. These pairs are marked "active" and we send a `GetOrdersPacket` for the newly activated pairs.

Other notable changes include:

- Tracking "active" pairs on a per-peer basis. "Active" pairs are those that have been advertised by a peer's node state, which we have been able to successfully perform a sanity swap for, and which have not encountered unexpected payment errors from our perspective.
- Not requesting all orders via a `GetOrdersPacket` automatically upon a successful connection and handshake.
- Ignoring order packets we receive for trading pairs that are not active.
- The `ListPeers` rpc call only returns active trading pairs.
- Sanity swaps can be disabled via the `nosanitychecks` config option.

BREAKING CHANGE: New SanitySwapPacket.

### CODE REVIEW

**Comment by erkarl** [comment]
Nice! I'm excited to get this feature in master as it will improve the likelihood of having real swaps fail. I went through the code once and got a pretty good overview of the implementation details - looks good on a high level. 👍 

This PR is quite big - I will do a more thorough/low-level review of the code later. Will also allocate time test this manually.

**Comment by erkarl** [comment]
Suggestions based on the weekly call:
- make the feature optional
- implement retry logic

**Comment by kilrau** [comment]
Can we get this one in?

**Comment by sangaman** [comment]
@erkarl @moshababo I've rebased and updated this PR so I think it's ready for review. Tests are currently failing due to a bug that I think came from #863 that causes the simulation tests to fail if `nosanitychecks` is disabled (which I want to do so that these sanity swaps get tested). I'm looking into fixing that in a separate PR.

**Comment by sangaman** [comment]
Will rebase on top of #901 when it's merged which should fix the tests.

**Comment by sangaman** [comment]
Rebased and passing the tests with sanity swaps enabled. @erkarl @moshababo

File: lib/orderbook/OrderBook.ts [hunk 1]
```diff
@@ -645,6 +672,73 @@ class OrderBook extends EventEmitter {
...
+      currenciesToVerify.add(quoteCurrency);
+    });
+
+    const verifiedCurrencies = new Set<string>();
+    const verifiedPairs: string[] = [];
+    const sanitySwapPromises: Promise<void>[] = [];
+
+    if (this.swaps) {
+      // Set a time limit for all sanity swaps to complete.
+      const sanitySwapTimeout = setTimeoutPromise(15000, false);
```
[the remark this answers is not in the document]
Comment (sangaman) [reply lib/orderbook/OrderBook.ts]: Yes and will also add a few code comments around this block.

**Pull request merged**